pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,400 @@
+            for i in range(len(rawWaveform))
+        ]
+        return timetags, waveform
+
+    def get_desc(self, descriptor=None):
+        """Get the descriptor of data being sent when querying device for waveform
+        Will decode a descriptor if provided one (Raw byte stream), otherwise it will query it
+        :return:
+        dict: A dictionnary with the keys:
+        - numDataPoints: the number of poitns in the waveform
+        - verticalGain: the voltage increment per code value (in V)
+        - verticalOffset: the voltage offset to add to the decoded voltage values
+        - horizInterval: the time interval between points in s
+        - horizOffset:the offset to add to the time steps
+        - descriptorOffset: Length of the C1:WF ALL,#9000000346 message
+
+        """
+        if descriptor is None:
+            command = "C{ch}:WF? DESC"
+            self.write(command)
+            descriptor = self.read_bytes(count=-1, break_on_termchar=True)
+        command = "WFSU?"
+        self.write(command)
+        wfsu = self.read()
+        sparsing = int(wfsu.split(",")[1])
+        number = int(wfsu.split(",")[3])
+        first = int(wfsu.split(",")[5])
+        descriptorOffset = 21
+        descriptorDictionnary = {"sparsing": sparsing, "numDataPoints": number, "firstPoint": first}
+        variables = [
+            ("commType", 32, "h"),
+            ("commOrder", 34, "h"),
+            ("waveDescLen", 36, "l"),
+            # ('numDataPoints',60,'l'),#This seems to give non-sense. A bug in the firmware?
+            ("pointsScreen", 120, "l"),
+            # ('firstPoint',124,'L'), #same
+            ("sparsingFactor", 136, "l"),
+            ("numSweeps", 148, "l"),
+            ("verticalGain", 156, "f"),
+            ("verticalOffset", 160, "f"),
+            ("horizInterval", 176, "f"),
+            ("horizOffset", 180, "d"),
+            ("acqDuration", 312, "f"),
+            ("recordType", 316, "h"),
+            ("processing", 318, "h"),
+            ("timebase", 324, "h"),
+            ("vertCoupling", 326, "h"),
+            ("probeAtt", 328, "f"),
+            ("fixedVertGain", 332, "h"),
+            ("bwLim", 334, "h"),
+        ]
+        for name, offset, format in variables:
+            (value,) = struct.unpack_from(
+                format,
+                descriptor,
+                offset=descriptorOffset + offset,
+            )
+            descriptorDictionnary.update({name: value})
+        if descriptor is None:
+            descriptorDictionnary.update({"descriptorOffset": descriptorOffset})
+        else:
+            descriptorDictionnary.update({"descriptorOffset": descriptorOffset + 346})
+
+        return descriptorDictionnary
+
+
+class TriggerChannel(Channel):
+    """
+    =========================================
+    Implementation of trigger control channel
+    =========================================
+    """
+
+    triggerConfDict = {}
```

Comment:
please use snake_case for variables and methods (i.e. all lower caps and separating words with underscores):
```suggestion
    trigger_config_dict
```

Please keep abbreviations consistent (here config for the methods below)